synapse/handlers/sso.py

```diff
@@ -720,11 +736,26 @@ async def handle_submit_username_request(
             request: HTTP request
             localpart: localpart requested by the user
             session_id: ID of the username mapping session, extracted from a cookie
+            use_display_name: whether the user wants to use the suggested display name
+            emails_to_use: emails that the user would like to use
         """
         session = self.get_mapping_session(session_id)
 
         # update the session with the user's choices
         session.chosen_localpart = localpart
+        session.use_display_name = use_display_name
+
+        session.emails_to_use = []
+        for email in emails_to_use:
+            # this is O(N^2), but N is small...
```

Comment:
This makes me wonder if they should both be sets (hopefully none of our code explodes if there's duplicates given!).

Reply:
hrm, duplicates are a thing I hadn't considered. I think I will make them sets.